compact.go

```diff
@@ -611,7 +651,8 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 				chks[i].Chunk = newChunk
 			}
 		}
-		if err := chunkw.WriteChunks(chks...); err != nil {
+		var err error
+		if chks, err = chunkw.WriteChunks(chks...); err != nil {
```

Comment:
let's not use `if _ = :...; err != nil` idiom then if we don't want to scope down those return variables.